test/unit/math/mix/core/operator_subtraction_test.cpp

```diff
@@ -6,3 +6,136 @@ TEST(mathMixCore, operatorSubtraction) {
+  stan::test::expect_ad_matvar(tols, f, matrix_m11, matrix_m11);
+}
+
+TEST(mathMixCore, operatorSubtractionMatrixZeroSize) {
+  auto f = [](const auto& x1, const auto& x2) {
+    return stan::math::subtract(x1, x2);
+  };
+  stan::test::ad_tolerances tols;
+  tols.hessian_hessian_ = 1e-1;
+  tols.hessian_fvar_hessian_ = 1e-1;
+  double scalar_a = 10;
+  Eigen::VectorXd vector_v0(0);
+  Eigen::RowVectorXd rowvector_rv0(0);
+  Eigen::MatrixXd matrix_m00(0, 0);
+  stan::test::expect_ad(f, scalar_a, vector_v0);
+  stan::test::expect_ad(f, vector_v0, scalar_a);
+  stan::test::expect_ad(f, scalar_a, rowvector_rv0);
+  stan::test::expect_ad(f, rowvector_rv0, scalar_a);
+  stan::test::expect_ad(f, scalar_a, matrix_m00);
+  stan::test::expect_ad(f, matrix_m00, scalar_a);
+  stan::test::expect_ad(f, matrix_m00, matrix_m00);
+
+  stan::test::expect_ad_matvar(f, scalar_a, vector_v0);
+  stan::test::expect_ad_matvar(f, vector_v0, scalar_a);
+  stan::test::expect_ad_matvar(f, scalar_a, rowvector_rv0);
+  stan::test::expect_ad_matvar(f, rowvector_rv0, scalar_a);
+  stan::test::expect_ad_matvar(f, scalar_a, matrix_m00);
+  stan::test::expect_ad_matvar(f, matrix_m00, scalar_a);
+  stan::test::expect_ad_matvar(f, matrix_m00, vector_v0);
+  stan::test::expect_ad_matvar(f, rowvector_rv0, vector_v0);
+  stan::test::expect_ad_matvar(f, vector_v0, rowvector_rv0);
+  stan::test::expect_ad_matvar(f, rowvector_rv0, matrix_m00);
+  stan::test::expect_ad_matvar(f, matrix_m00, matrix_m00);
+}
+
+TEST(mathMixCore, operatorSubtractionMatrixNormal) {
+  auto f = [](const auto& x1, const auto& x2) {
+    return stan::math::subtract(x1, x2);
+  };
+  stan::test::ad_tolerances tols;
+  tols.hessian_hessian_ = 1e-1;
+  tols.hessian_fvar_hessian_ = 1e-1;
+  double scalar_a = 10;
+  Eigen::VectorXd vector_v(2);
+  vector_v << 100, -3;
+  Eigen::RowVectorXd rowvector_rv(2);
+  rowvector_rv << 100, -3;
+  Eigen::MatrixXd matrix_m(2, 2);
+  matrix_m << 100, 0, -3, 4;
+  stan::test::expect_ad(tols, f, scalar_a, vector_v);
+  stan::test::expect_ad(tols, f, vector_v, scalar_a);
+  stan::test::expect_ad(tols, f, scalar_a, rowvector_rv);
+  stan::test::expect_ad(tols, f, rowvector_rv, scalar_a);
+  stan::test::expect_ad(tols, f, scalar_a, matrix_m);
+  stan::test::expect_ad(tols, f, matrix_m, matrix_m);
+  stan::test::expect_ad(tols, f, matrix_m, scalar_a);
+
+  stan::test::expect_ad_matvar(tols, f, scalar_a, vector_v);
+  stan::test::expect_ad_matvar(tols, f, vector_v, scalar_a);
+  stan::test::expect_ad_matvar(tols, f, scalar_a, rowvector_rv);
+  stan::test::expect_ad_matvar(tols, f, rowvector_rv, scalar_a);
+  stan::test::expect_ad_matvar(tols, f, rowvector_rv, vector_v);
+  stan::test::expect_ad_matvar(tols, f, vector_v, rowvector_rv);
+  stan::test::expect_ad_matvar(tols, f, scalar_a, matrix_m);
+  stan::test::expect_ad_matvar(tols, f, matrix_m, scalar_a);
+  stan::test::expect_ad_matvar(tols, f, matrix_m, vector_v);
+  stan::test::expect_ad_matvar(tols, f, rowvector_rv, matrix_m);
```

Comment:
These should definitely just be throwing errors. I guess it's good to have them so the error checks between `mat<var>` and `var<mat>` are consistent. Probably helpful to put them off to the side and say "errors".

Reply:
I put a space there to break them out